UnityClient/Assets/Scenes/Map/MapUiController.cs

```diff
@@ -102,4 +107,27 @@ public class MapUiController : MonoBehaviour {
     public void HideTooltip() {
         Tooltip.SetText(null, Vector3.zero);
     }
+
+    public void OnRestartAnswer(ushort cmd, int size, InPacket packet) {
+        if (packet is ZC.RESTART_ACK) {
+            var pkt = packet as ZC.RESTART_ACK;
```

Comment:
This can be shortened to `if (packet is ZC.RESTART_ACK pkt)`. Just have in mind that this name can't be repeated in else-if situations